numpy/typing/_array_like.py

```diff
@@ -0,0 +1,27 @@
+import sys
+from typing import Any, overload, Sequence, Tuple, Union
+
+from numpy import ndarray
+from ._dtype_like import DtypeLike
+
+if sys.version_info >= (3, 8):
```

Comment:
This is the "no hard dependency on `typing_extensions` dance mentioned in the PR description.

Reply:
I'd worry about that because it's pretty common for projects to run their tests suites with warnings turned into errors, and I could see something like e.g. a SciPy test run that doesn't have mypy installed and errors out because of the warning.

Reply:
Hmm, that might be a problem, yes.
Does NumPy have a logger where where such information could be displayed? Since I suspect that silently setting `_SupportsArray` to `Any` (and by extension `ArrayLike`) could result in some unexpected issues (at least from a end user perspective).

If not, then it should be mentioned in the documentation.

Reply:
No logger, when most of the action is happening in C code (potentially with the GIL released) logging doesn't work great. I added a big warning to to the top of the documentation in https://github.com/numpy/numpy/pull/16558/commits/4a120f0d073506329b564bfe9e69db2f6b612964.

Reply:
Ah, so be it.
At least it is documented now, which is the most important thing.